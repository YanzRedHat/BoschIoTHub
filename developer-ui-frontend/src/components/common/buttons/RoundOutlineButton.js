/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */

import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { OutlineBtn, Icon } from "./OutlineButton";
import { getThemeColor } from "utils";
// SVG Imports
import SubmitIcon from "images/submitCaret.svg";

const RoundOutlineBtn = OutlineBtn.extend`
  border-radius: 2.034rem;

  svg.submit-icon {
    path {
      fill: ${props => getThemeColor(props)};
    }
  }

  &:disabled {
    svg.submit-icon path {
      fill: ${props => props.theme.disabledFontColor};
    }
  }

  &:hover {
    border-style: unset;
    &:not(:disabled) {
      padding: calc(0.65em + 0.125em) calc(1.1em + 0.125em)
        calc(0.65em + 0.125em) calc(1.1em + 0.125em);
    }
    svg.submit-icon {
      transform: translateX(0.5em);
      path {
        fill: #fff;
      }
    }
  }
  &:disabled {
    &:hover {
      border-style: solid;
      svg.submit-icon {
        transform: translateX(0);
        path {
          transform: translateX(0);
          fill: ${props => props.theme.disabledFontColor};
        }
      }
    }
  }
`;

const SubmitSvg = styled(SubmitIcon)`
  transition: transform 0.3s ease-out;
  padding-left: 0.75em;
  height: 0.6em;
  path {
    transition: fill 0.5s ease-out;
  }
`;

const RoundOutlineButton = props => {
  const { children, icon, submitAnimation, primary, ...other } = props;
  return (
    <RoundOutlineBtn primary={primary ? 1 : 0} {...props}>
      <span>
        {children} {icon ? <Icon {...props} /> : null}{" "}
        {submitAnimation && <SubmitSvg className="submit-icon" {...other} />}
      </span>
    </RoundOutlineBtn>
  );
};

RoundOutlineButton.propTypes = {
  children: PropTypes.string,
  primary: PropTypes.bool,
  cancel: PropTypes.bool,
  secondary: PropTypes.bool,
  danger: PropTypes.bool,
  icon: PropTypes.element,
  submitAnimation: PropTypes.bool
};

export default RoundOutlineButton;
