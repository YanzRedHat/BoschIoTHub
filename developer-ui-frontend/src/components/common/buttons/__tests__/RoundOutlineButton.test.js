/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import { shallow } from "enzyme";
import RoundOutlineButton from "components/common/buttons/RoundOutlineButton.js";
import CopyIcon from "images/copyClipboardIcon.svg";

describe("RoundOutlineButton", () => {
  it("should be defined", () => {
    expect(RoundOutlineButton).toBeDefined();
  });
  it("should render correctly", () => {
    const tree = shallow(<RoundOutlineButton name="button test" />);
    expect(tree).toMatchSnapshot();
  });
  it("should have a prop primary", () => {
    const tree = shallow(<RoundOutlineButton name="button test" primary />);
    expect(tree.props().primary).toEqual(true);
  });
  it("should have a prop secondary", () => {
    const tree = shallow(<RoundOutlineButton name="button test" secondary />);
    expect(tree.props().secondary).toEqual(true);
  });
  it("should have a prop danger", () => {
    const tree = shallow(<RoundOutlineButton name="button test" danger />);
    expect(tree.props().danger).toEqual(true);
  });
  it("should have a prop cancel", () => {
    const tree = shallow(<RoundOutlineButton name="button test" cancel />);
    expect(tree.props().cancel).toEqual(true);
  });
  it("should have a prop submitAnimation", () => {
    const tree = shallow(
      <RoundOutlineButton name="button test" submitAnimation />
    );
    expect(tree.props().submitAnimation).toEqual(true);
  });
  it.skip("should have a prop icon", () => {
    const tree = shallow(
      <RoundOutlineButton name="button test" icon={CopyIcon} />
    );
    expect(tree.props().icon).toEqual(CopyIcon);
  });
  it("should call mock function when button is clicked", () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <RoundOutlineButton name="button test" onClick={mockFn} />
    );
    tree.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
  it.skip("Compared values have no visual difference but may have different properties", () => {
    const tree = shallow(
      <RoundOutlineButton name="button test" children={CopyIcon} />
    );
    expect(JSON.stringify(tree.props().children)).toEqual(
      JSON.stringify(<span>{CopyIcon} </span>)
    );
  });
});
