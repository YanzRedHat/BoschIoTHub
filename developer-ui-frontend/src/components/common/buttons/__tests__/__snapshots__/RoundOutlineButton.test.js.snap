// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundOutlineButton should render correctly 1`] = `
<styled.button
  name="button test"
  primary={0}
>
  <span>
     
     
  </span>
</styled.button>
`;
